nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -65,3 +73,234 @@ def run(self, data, commit):  # pylint: disable=too-many-locals
+        else:
+            cvss_base_score = cve_impact['baseMetricV2']['cvssV2']['baseScore']
+            cvss_severity = cve_impact['baseMetricV2']['severity']
+            cvssv2_score = cve_impact['baseMetricV2']['exploitabilityScore']                           
+            cvssv3_score = None
+
+        all_cve_info = {
+            'url': cve_url,
+            'description': cve_description,
+            'published_date': cve_published_date,
+            'modified_date': cve_modified_date,
+            'cvss_base_score': cvss_base_score,
+            'cvss_severity': cvss_severity,
+            'cvssv2_score': cvssv2_score,
+            'cvssv3_score': cvssv3_score
+            }
+
+        return all_cve_info
+        
+    def get_cve_info(self, cpe_software_search_url: str, software_id=None) -> dict:
+        """Search NIST for software and related CVEs"""
+        cpe_info = json.loads(requests.get(cpe_software_search_url).text)
+        all_cve_info = {}
+        if len(cpe_info['result']['cpes']) > 0:
+            cve_list = cpe_info['result']['cpes'][0].get('vulnerabilities', [])
+            base_url = "https://services.nvd.nist.gov/rest/json/cve/1.0/"
+
+            dlc_cves = [cve.name for cve in CVELCM.objects.all()]
+            
+            if len(cve_list) > 0:
+                for cve in cve_list:
+                    if cve not in dlc_cves:
+                        if re.search("^CVE", cve):
+                            all_cve_info[cve] = self.prep_cve_for_dlc(base_url+cve)
+                            sleep(.25)
+                    else:
+                        existing_cve = CVELCM.objects.get(name=cve)
+                        self.associate_software_to_cve(software_id, existing_cve.id)
+
+        return all_cve_info
+
+    def create_dlc_cves(self, software_id, cpe_cves):
+        """Create the list of items that will need to be inserted to DLC CVEs"""
+        dlc_cves = CVELCM.objects.all()
+        for cve, info in cpe_cves.items():
+            if cve not in dlc_cves:
+                create_cves = CVELCM.objects.get_or_create(
+                    name=cve, 
+                    description=(
+                        f"{info['description'][0:251]}..." \
+                            if len(info['description']) > \
+                                255 else info['description']),
+                    published_date=date.fromisoformat(info['published_date'][0:10]),
+                    last_modified_date=date.fromisoformat(info['modified_date'][0:10]),
+                    link=info['url'],
+                    cvss=info['cvss_base_score'],
+                    severity=info['cvss_severity'],
+                    cvss_v2=info['cvssv2_score'],
+                    cvss_v3=info['cvssv3_score'],
+                    comments="ENTRY CREATED BY NAUTOBOT NIST JOB"
+                    )
+
+                self.log_info(message=f"""Created {cve}""")
+                cve = CVELCM.objects.get(name=cve)
```

Comment:
This is redundant as `get_or_create()` will have returned this `CVELCM` instance already.